crates/biome_js_analyze/src/services/semantic_class.rs

```diff
@@ -817,6 +868,44 @@ fn is_within_scope_without_shadowing(
     false
 }
 
+pub fn is_meaningful_read(node: &MeaningfulReadNode) -> Option<bool> {
+    is_used_in_expression_context(node)
+}
+
+/// If the node is a parenthesized expression, return the inner expression,
+/// otherwise return the node itself.
+fn skip_parentheses(node: JsSyntaxNode) -> JsSyntaxNode {
+    if node.kind() == JsSyntaxKind::JS_PARENTHESIZED_EXPRESSION {
+        // Assume the first child is the expression inside parentheses
+        if let Some(inner) = node.first_child() {
+            return inner;
+        }
+    }
+    node
+}
+
+fn is_used_in_expression_context(node: &MeaningfulReadNode) -> Option<bool> {
```

Comment:
You must document this function. Its name isn't descriptive enough. For example what's "context"? What's the business logic? What are the criteria? There's a lot to explain